iml-manager-cli/src/update_repo_file.rs

```
@@ -0,0 +1,111 @@
+// Copyright (c) 2019 DDN. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+use crate::{
+    api_utils::{create_command, get_hosts, run_cmd, wait_for_cmd, SendCmd, SendJob},
+    display_utils::{display_cancelled, display_cmd_state, display_error, start_spinner},
+};
+use futures::{
+    future::{self, Either},
+    Future,
+};
+use iml_wire_types::Host;
+use std::collections::BTreeSet;
+use std::process::exit;
+use structopt::StructOpt;
+
+#[derive(StructOpt, Debug)]
+pub struct UpdateRepoFileHosts {
+    /// The host(s) to update. Takes a hostlist expression
+    #[structopt(short = "h", long = "hosts")]
+    hosts: String,
+}
+
+fn filter_known_hosts<'a, 'b>(
+    hostlist: &'b BTreeSet<String>,
+    api_hosts: &'a Vec<Host>,
+) -> Vec<&'a Host> {
+    api_hosts
+        .iter()
+        .filter(move |x| hostlist.contains(&x.fqdn) || hostlist.contains(&x.nodename))
+        .collect()
+}
+
+fn has_host(api_hosts: &Vec<&Host>, host: &String) -> bool {
+    api_hosts
+        .iter()
+        .any(|y| &y.fqdn == host || &y.nodename == host)
+}
+
+#[derive(serde::Serialize)]
+pub struct HostId {
+    host_id: u32,
+}
+
+pub fn update_repo_file_cli(config: UpdateRepoFileHosts) -> () {
+    let r = hostlist_parser::parse(&config.hosts).expect("Couldn't parse hostlist expression.");
```

Nit, would be better to return the error here.